test/opf/sdpwrm.jl

```diff
@@ -0,0 +1,227 @@
+using LinearAlgebra
+using SparseArrays
+
+function test_opf_pm(::Type{OPFGenerator.SDPOPF}, data::Dict)
+    OPF = OPFGenerator.SDPOPF
+
+    data["basic_network"] || error("Input data must be in basic format to test")
+    N = length(data["bus"])
+    E = length(data["branch"])
+    G = length(data["gen"])
+
+    # Solve OPF with PowerModels
+    solver = OPT_SOLVERS[OPF]
+    res_pm = PM.solve_opf(data, PM.SDPWRMPowerModel, solver)
+
+    # Build and solve OPF with OPFGenerator
+    solver = OPT_SOLVERS[OPF]
+    opf = OPFGenerator.build_opf(OPF, data, solver)
+    OPFGenerator.solve!(opf)
+    res = OPFGenerator.extract_result(opf)
+
+    # Check that the right problem was indeed solved
+    @test res["meta"]["formulation"] == string(OPF)
+    @test res["meta"]["termination_status"] ∈ ["LOCALLY_SOLVED", "OPTIMAL"]
+    @test res["meta"]["primal_status"] == "FEASIBLE_POINT"
+    @test res["meta"]["dual_status"] == "FEASIBLE_POINT"
+    # Check objective value against PowerModels
+    @test isapprox(res["meta"]["primal_objective_value"], res_pm["objective"], atol=1e-3, rtol=1e-3)
+    @test isapprox(res["meta"]["primal_objective_value"], res["meta"]["dual_objective_value"], rtol=1e-6)
+
+    # Force PM solution into our model, and check that the solution is feasible
+    # TODO: use JuMP.primal_feasibility_report instead
+    #    (would require extracting a variable => value Dict)
+    sol_pm = res_pm["solution"]
+    var2val_pm = Dict(
+        :pg => Float64[
+            get(get(sol_pm["gen"], "$g", Dict()), "pg", 0) for g in 1:G
+        ],
+        :qg => Float64[
+            get(get(sol_pm["gen"], "$g", Dict()), "qg", 0) for g in 1:G
+        ],
+        # The diagonal elements of sol_pm["WR"]. Note that the rows and columns of sol_pm["WR"] are ordered in the order of ref[:bus].
+        :w => Float64[sol_pm["bus"]["$i"]["w"] for i in 1:N]
+    )
+    model = opf.model
+    for varname in [:pg, :qg]
+        x = model[varname]
+        v = var2val_pm[varname]
+        fix.(x, v; force=true)
+    end
+    fix.(diag(model[:WR]), var2val_pm[:w]; force=true)
+
+    optimize!(model)
+    @test termination_status(model) ∈ [OPTIMAL, ALMOST_OPTIMAL, LOCALLY_SOLVED, ALMOST_LOCALLY_SOLVED]
+    @test primal_status(model) ∈ [FEASIBLE_POINT, NEARLY_FEASIBLE_POINT]
+    # Also check that we get the same objective value as PowerModels
+    @test isapprox(objective_value(opf.model), res_pm["objective"], atol=1e-3, rtol=1e-3)
+
+    return nothing
+end
+
+"""
+    _test_sdpwrm_DualFeasibility()
+
+Test dual feasibility of SDPWRM problem.
+
+This test is executed on the 14 bus system.
+"""
+function _test_sdpwrm_DualFeasibility()
```

Comment:
TODO @mtanneau : 
* add a 5-pjm test case where we artificially add a branch to have parallel branches
* also make sure that branches `(i, j)` with `i > j` (e.g. (4, 1) are supported as expected